plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogParser.java

```diff
@@ -157,7 +180,35 @@ public static Object deserializePartitionValue(DeltaLakeColumnHandle column, Opt
 
     private static Long readPartitionTimestamp(String timestamp)
     {
-        ZonedDateTime zonedDateTime = LocalDateTime.parse(timestamp, PARTITION_TIMESTAMP_FORMATTER).atZone(UTC);
+        ParsePosition position = new ParsePosition(0);
+        TemporalAccessor accessor = PARTITION_TIMESTAMP_FORMATTER.parseUnresolved(timestamp, position);
+        ZonedDateTime zonedDateTime;
+        if (position.getErrorIndex() == -1 && timestamp.length() == position.getIndex()) {
+            if (accessor.isSupported(NANO_OF_SECOND)) {
```

Comment:
These conditions look hard to understand for me. Could you leave a code comment with the example?